core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -52,7 +55,22 @@ import org.apache.spark.storage._
 private[spark] object JsonProtocol {
   // TODO: Remove this file and put JSON serialization into each individual class.
 
-  private implicit val format = DefaultFormats
+  // Events defined in other modules are allowed to be registered through implementing the
+  // SparkListenerEventRegister trait. These events are written to Json using Jackson.
+  val eventRegisters: Iterable[SparkListenerEventRegister] = {
+    val loader = Utils.getContextOrSparkClassLoader
+    ServiceLoader.load(classOf[SparkListenerEventRegister], loader).asScala
+  }
+
+  var eventClasses = new ListBuffer[Class[_]]()
```

Comment:
This can be a `val`. I still want to check how this stuff is used, though (haven't made it to that part of the patch yet).

    val eventClasses = eventRegisters.flatMap { r => eventRegister.getEventClasses() }.toSeq